src/cli.rs

```diff
@@ -731,99 +719,132 @@ impl Opts {
+                        Clear(ClearType::FromCursorDown)
+                    )
+                    .unwrap();
+                }
 
-                pin_writer(style.apply(name), status, i)?;
+                in_progress.insert(name);
+                for n in in_progress.iter() {
+                    stderr.queue(Print(n.dark_yellow())).unwrap();
+                    stderr.write_all(b"\n").unwrap();
+                }
+                write!(stderr, "{}/{length} Pins", finished.get()).unwrap();
 
+                stderr.flush().unwrap();
+            })
+            .map(|(name, pin)| async move {
+                let diff = Self::update_one(pin, strategy).await?;
                 anyhow::Result::<_, anyhow::Error>::Ok((name, diff))
             });
 
         stream::iter(update_iter)
             .buffer_unordered(opts.max_concurrent_downloads)
-            .try_filter(|(_, diff)| future::ready(diff.is_empty().not()))
-            .try_collect::<Vec<_>>()
-            .await?
-            .into_iter()
-            .for_each(|(name, diff)| print_diff(name, diff));
+            .try_for_each(|(name, diff)| {
+                fn write_diff(writer: &mut impl Write, name: &str, diff: Vec<diff::DiffEntry>) {
+                    if diff.is_empty() {
+                        writeln!(writer, "[{name}] No Changes").unwrap();
+                    } else {
+                        writeln!(writer, "[{name}] Changes:").unwrap();
+                        for entry in diff {
+                            write!(writer, "{entry}").unwrap();
+                        }
+                    }
+                }
+
+                let mut stderr = stderr().lock();
+                if stderr.is_terminal().not() {
+                    write_diff(&mut stderr, name, diff);
+                    return future::ready(Ok(()));
+                }
+                let mut in_progress = in_progress.borrow_mut();
+
+                queue!(
+                    stderr,
+                    MoveToPreviousLine(in_progress.len() as u16),
+                    Clear(ClearType::FromCursorDown)
+                )
+                .unwrap();
+
+                in_progress.remove(name.as_str());
+                write_diff(&mut stderr, name, diff);
+
+                for n in in_progress.iter() {
+                    stderr.queue(Print(n.dark_yellow())).unwrap();
+                    stderr.write_all(b"\n").unwrap();
+                }
+                finished.set(finished.get() + 1);
+                write!(stderr, "{}/{length} Pins", finished.get()).unwrap();
+
+                stderr.flush().unwrap();
+                future::ready(Ok(()))
+            })
+            .await?;
+        if length != 0 && stderr().is_terminal() {
+            eprintln!();
+        }
 
         if !opts.dry_run {
             self.write_pins(&pins)?;
-            log::info!("Update successful.");
+            log::info!("Updated lockfile.");
```

Comment:
```suggestion
            log::info!("Updated successful.");
```

Any strong reason for that change? Otherwise I'd prefer it the current way

Reply:
I think it makes it clearer what the condition is behind the log, but that's not really strong reasoning since users should only really care about this if they are specifically thinking about dry runs.